pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -584,3 +593,163 @@ func (a *Actuator) nodeAddresses(host *bmh.BareMetalHost) ([]corev1.NodeAddress,
 
 	return addrs, nil
 }
+
+//deleteRemediationAnnotations deletes remediation-in-progress and remediation strategy annotations
+func (a *Actuator) deleteRemediationAnnotations(ctx context.Context, machine *machinev1.Machine) error {
+	if len(machine.Annotations) == 0 {
+		return nil
+	}
+
+	delete(machine.Annotations, remediationInProgressAnnotation)
+	delete(machine.Annotations, externalRemediationAnnotation)
+
+	if err := a.client.Update(ctx, machine); err != nil {
+		log.Printf("Failed to delete annotations of Machine: %s", machine.Name)
+		return err
+	}
+
+	return nil
+}
+
+//hasRebootAnnotation checks if the reboot annotation exist on the baremetalhost
+func hasRebootAnnotation(baremetalhost *bmh.BareMetalHost) (exists bool) {
+	if len(baremetalhost.Annotations) > 0 {
+		_, exists = baremetalhost.Annotations[rebootAnnotation]
+	}
+	return
+}
+
+//addRemediationInProgressAnnotation adds a remediation-in-progress annotation to the machine
+func (a *Actuator) addRemediationInProgressAnnotation(ctx context.Context, machine *machinev1.Machine) error {
+	if machine.Annotations == nil {
+		machine.Annotations = make(map[string]string)
+	}
+
+	machine.Annotations[remediationInProgressAnnotation] = ""
+
+	err := a.client.Update(ctx, machine)
+	if err != nil {
+		log.Printf("Failed to add remediation in progess annotation to %s: %s", machine.Name, err.Error())
+	}
+
+	return err
+}
+
+//requestPowerOn removes reboot annotation on baremetalhost which signal BMO to power on the machine
+func (a *Actuator) requestPowerOff(ctx context.Context, baremetalhost *bmh.BareMetalHost) error {
+	if baremetalhost.Annotations == nil {
+		baremetalhost.Annotations = make(map[string]string)
+	}
+
+	baremetalhost.Annotations[rebootAnnotation] = ""
```

Comment:
I wonder if this would all be easier to follow if this annotation was called `requestPowerOffAnnotation`. Or if this function was called `requestReboot`. Either might help readability of the `remediateIfNeeded` function.

Reply:
I agree with that `requestPowerOffAnnotation` is better than `rebootAnnotation`. I'll change this.
`requestReboot` is not a proper name for this function, since it is actually requesting a power off, and not a reboot.
The power on will happen when this annotation will be removed.

Reply:
Changed to `requestPowerOffAnnotation`